lib/ansible/modules/cloud/ovh/ovh_dns.py

```diff
@@ -0,0 +1,312 @@
+options:
+    domain:
+        required: true
+        description:
+            - Name of the domain zone
+    name:
+        required: true
+        description:
+            - Name of the DNS record
+    value:
+        required: true
+        description:
+            - Value of the DNS record (i.e. what it points to)
+    type:
+        default: A
+        choices: ['A', 'AAAA', 'CNAME', 'DKIM', 'LOC', 'MX', 'NAPTR', 'NS', 'PTR', 'SPF', 'SRV', 'SSHFP', 'TXT']
+        description:
+            - Type of DNS record (A, AAAA, PTR, CNAME, etc.)
+    ttl:
+        default: 0
+        description:
+            - Time to live of the DNS record
+    state:
+        default: present
+        choices: ['present', 'absent']
+        description:
+            - Determines wether the record is to be created/modified or deleted
+    endpoint:
+        required: true
+        description:
+            - The endpoint to use ( for instance ovh-eu)
+    application_key:
+        required: true
+        description:
+            - The applicationKey to use
+    application_secret:
+        required: true
+        description:
+            - The application secret to use
+    consumer_key:
+        required: true
+        description:
+            - The consumer key to use
+'''
+
+EXAMPLES = '''
+# Create a typical A record
+- ovh_dns:
+    state: present
+    domain: mydomain.com
+    name: db1
+    value: 10.10.10.10
+    ttl: 3600
+    endpoint: ovh-eu
+    application_key: yourkey
+    application_secret: yoursecret
+    consumer_key: yourconsumerkey
+
+# Create a CNAME record
+- ovh_dns:
+    state: present
+    domain: mydomain.com
+    name: dbprod
+    type: CNAME
+    value: db1
+    ttl: 3600
+    endpoint: ovh-eu
+    application_key: yourkey
+    application_secret: yoursecret
+    consumer_key: yourconsumerkey
+
+# Delete an existing record, must specify all parameters
+- ovh_dns:
+    state: absent
+    domain: mydomain.com
+    name: dbprod
+    type: CNAME
+    value: db1
+    endpoint: ovh-eu
+    application_key: yourkey
+    application_secret: yoursecret
+    consumer_key: yourconsumerkey
+'''
+
+RETURN='''
+'''
+
+import os
+import sys
+
+try:
+    import ovh
+    from ovh.exceptions import APIError
+    HAS_OVH=True
+except ImportError:
+    HAS_OVH=False
+
+def get_ovh_client(module):
+    endpoint = module.params.get('endpoint')
+    application_key = module.params.get('application_key')
+    application_secret = module.params.get('application_secret')
+    consumer_key = module.params.get('consumer_key')
+
+    return ovh.Client(
+        endpoint=endpoint,
+        application_key=application_key,
+        application_secret=application_secret,
+        consumer_key=consumer_key
+    )
+
+
+def get_domain_records(client, domain):
+    """Obtain all records for a specific domain"""
+    records = {}
+
+    # List all ids and then get info for each one
+    record_ids = client.get('/domain/zone/{}/record'.format(domain))
```

Comment:
Format replacement strings with nothing inside of them, {}, are only available in python-2.7 or better. So you'll need to make sure that all your formats have numbers inside of them. For example, this line would then be:
`record_ids = client.get('/domain/zone/{0}/record'.format(domain))`